akka-cluster-tools/src/main/scala/akka/cluster/singleton/ClusterSingletonManager.scala

```diff
@@ -898,6 +1046,28 @@ class ClusterSingletonManager(
     case WasOldest → _      ⇒ cancelTimer(TakeOverRetryTimer)
   }
 
+  onTransition {
+    case (AcquiringLease, to) if to != Oldest ⇒
+      stateData match {
+        case AcquiringLeaseData(true, _) ⇒
+          logInfo("Releasing lease as leaving AcquiringLease going to {}", to)
+          import context.dispatcher
+          lease.foreach(l ⇒ pipe(l.release().map[Any](ReleaseLeaseResult).recover {
+            case NonFatal(t) ⇒ ReleaseLeaseFailure(t)
```

Comment:
`NonFatal` not needed here since that is taken care of by scala Future

Reply:
Doh